frontend/src/metabase/containers/SaveQuestionModal.jsx

```diff
@@ -104,17 +104,21 @@ export default class SaveQuestionModal extends Component {
         ? t`First, save your question`
         : t`First, save your model`;
 
-    const singleStepTitle =
-      questionType === "question" ? t`Save question` : t`Save model`;
-
-    const title = this.props.multiStep ? multiStepTitle : singleStepTitle;
-
     const showSaveType =
       !card.id &&
       !!originalCard &&
       !originalCard.dataset &&
       originalCard.can_write;
 
+    const singleStepTitle =
+      questionType === "question"
+        ? showSaveType
+          ? t`Save question`
```

Comment:
Not sure how to do this via if-elses when the conditional is within the instantiation of a const like this. I'm sure there's a better pattern?

Reply:
We usually extract it to a function with multiple returns to keep using const everywhere

Reply:
@ranquild KK, thanks. I'm sure I'm doing this wrong 'cause I hit a server error when I try to save now, but am I headed in the right general direction with this? https://gist.github.com/mazameli/6ae5b8d7f4e06119d79014f43019c1a1

Reply:
@mazameli Something like this should work. I'd put this function outside of the component and call it here.

```
const getSingleStepTitle = (questionType, showSaveType) => {
  if (questionType === "model") {
    return t`Save model`;
  } else if (showSaveType) {
    return t`Save question`;
  } else {
    return t`Save new question`;
  }
};

```